modules/dnn/src/model.cpp

```diff
@@ -0,0 +1,229 @@
+    std::tie(classId, conf) = classify(frame);
+}
+
+DetectionModel::DetectionModel(const String& model, const String& config)
+    : Model(model, config) {};
+
+DetectionModel::DetectionModel(const Net& network) : Model(network) {};
+
+Ptr<DetectionModel> DetectionModel::create(const String& model, const String& config) {
+    return makePtr<DetectionModel>(model, config);
+}
+
+void DetectionModel::detect(InputArray frame, CV_OUT std::vector<int>& classIds,
+                   CV_OUT std::vector<float>& confidences, CV_OUT std::vector<Rect>& boxes,
+                   float confThreshold, float nmsThreshold)
+{
+    std::vector<Mat> detections;
+    impl->predict(*this, frame.getMat(), detections);
+
+    boxes.clear();
+    confidences.clear();
+    classIds.clear();
+
+    int frameWidth  = frame.cols();
+    int frameHeight = frame.rows();
+
+    std::vector<String> layerNames = getLayerNames();
+    int lastLayerId = getLayerId(layerNames.back());
+    Ptr<Layer> lastLayer = getLayer(lastLayerId);
+
+    std::vector<int> predClassIds;
+    std::vector<Rect> predBoxes;
+    std::vector<float> predConf;
+    if (lastLayer->type == "DetectionOutput") {
+        // Network produces output blob with a shape 1x1xNx7 where N is a number of
+        // detections and an every detection is a vector of values
+        // [batchId, classId, confidence, left, top, right, bottom]
+        for (int i = 0; i < detections.size(); ++i)
+        {
+            float* data = (float*)detections[i].data;
+            for (int j = 0; j < detections[i].total(); j += 7)
+            {
+                float conf = data[j + 2];
+                if (conf < confThreshold)
+                    continue;
+
+                float left   = data[j + 3];
+                float top    = data[j + 4];
+                float right  = data[j + 5];
+                float bottom = data[j + 6];
+                float width  = right  - left;
+                float height = bottom - top ;
+
+                if (width * height <= 1)
+                    boxes.emplace_back(left * frameWidth, top * frameHeight,
+                                       width * frameWidth + 1, height * frameHeight + 1);
+                else
+                    boxes.emplace_back(left, top, width + 1, height + 1);
+
+                classIds.push_back(static_cast<int>(data[j + 1]));
+                confidences.push_back(conf);
+            }
+        }
+    } else if (lastLayer->type == "Region") {
+        for (int i = 0; i < detections.size(); ++i) {
+            // Network produces output blob with a shape NxC where N is a number of
+            // detected objects and C is a number of classes + 4 where the first 4
+            // numbers are [center_x, center_y, width, height]
+            float* data = (float*)detections[i].data;
+            for (int j = 0; j < detections[i].rows; ++j, data += detections[i].cols) {
+
```

Comment:
Very strange line break. What about 
```cpp
for (int j = 0; j < detections[i].rows; ++j, data += detections[i].cols)
{
    // ...
```
?

Please also fix it in other places. We mostly use opening parentheses on separate lines.